src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1837,7 +1840,7 @@ private void AddServicesProducedInfo(IResource modelResource, IAnnotationHolder
                         $"The endpoint '{ea.Name}' for resource '{modelResourceName}' requested a proxy ({nameof(ea.IsProxied)} is true). Non-container resources cannot be proxied when both {nameof(ea.TargetPort)} and {nameof(ea.Port)} are specified with the same value.");
                 }
 
-                if (appResource.DcpResource is ExecutableReplicaSet && ea.TargetPort is int)
+                if (HasMultipleReplicas(appResource.DcpResource) && ea.TargetPort is int)
```

Comment:
I'm not sure about this change. Why did this throw if the resource could have replicas rather than check if there are multiple replicas?

Reply:
I do not think an ApplicationExecutor change or DCP custom annotation is necessary here. The problem should be easy to detect at Aspire model level. 

The gist of the check is that if someone has more than one instance of a program that uses a "proxy-less" endpoint, that will surely end up in a port conflict.

Reply:
James and I chatted. This is fine for now, and will help if ExecutableReplicaSets are used again someday.

Reply:
We should just use the ModelResource. It's much simpler 😄